certbot-apache/certbot_apache/tests/parser_test.py

```diff
@@ -21,6 +22,18 @@ def tearDown(self):
         shutil.rmtree(self.config_dir)
         shutil.rmtree(self.work_dir)
 
+    def test_bad_parse(self):
+        self.parser.parse_file(os.path.join(self.parser.root,
+                                            "conf-available", "bad_conf_file.conf"))
+        self.assertRaises(
+            errors.PluginError, self.parser.check_parsing_errors, "httpd.aug")
+
+    def test_bad_save(self):
+        mock_save = mock.Mock()
+        mock_save.side_effect = IOError
+        self.config.parser.aug.save = mock_save
```

Comment:
nit: If we're moving this test to the parser, can we just call `parser.unsaved_files` directly?